bson/bsonoptions/struct_codec_options.go

```diff
@@ -6,12 +6,15 @@
 
 package bsonoptions
 
+var defaultAllowOverwritingInlinedFields = true
+
 // StructCodecOptions represents all possible options for struct encoding and decoding.
 type StructCodecOptions struct {
-	DecodeZeroStruct        *bool // Specifies if structs should be zeroed before decoding into them. Defaults to false.
-	DecodeDeepZeroInline    *bool // Specifies if structs should be recursively zeroed when a inline value is decoded. Defaults to false.
-	EncodeOmitDefaultStruct *bool // Specifies if default structs should be considered empty by omitempty. Defaults to false.
-	AllowUnexportedFields   *bool // Specifies if unexported fields should be marshaled/unmarshaled. Defaults to false.
+	DecodeZeroStruct              *bool // Specifies if structs should be zeroed before decoding into them. Defaults to false.
+	DecodeDeepZeroInline          *bool // Specifies if structs should be recursively zeroed when a inline value is decoded. Defaults to false.
+	EncodeOmitDefaultStruct       *bool // Specifies if default structs should be considered empty by omitempty. Defaults to false.
+	AllowUnexportedFields         *bool // Specifies if unexported fields should be marshaled/unmarshaled. Defaults to false.
+	AllowOverwritingInlinedFields *bool // Specifies if fields in inlined structs can be overwritten by higher level struct fields with the same key. Defaults to true.
```

Comment:
Very minor nit: based on the names of the existing options, I don't think this should start with the prefix "Allow". The only other (`AllowUnexportedFields`) describes a property of the struct, not a behavior of the encoder/decoder. I'd suggest renaming to `OverwriteDuplicatedInlinedFields` or `AllowDuplicatedInlinedFields` if you want to stick with this prefix.